ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -435,6 +462,57 @@ private byte[] serializeValue(
     }
   }
 
+  /**
+   * Add missing required fields to the passed {@code FormatInfo} and return a
+   * new {@code FormatInfo} with the new fields. These fields are required to serialize the
+   * key and value schema correctly. For instance, if running INSERT on a stream with a SR schema
+   * name different from the default name used in Connect (i.e. ConnectDefault1 for Protobuf).
+   * </p>
+   * Note: I initially thought of injecting the SCHEMA_FULL_NAME property in the WITH clause
+   * when creating the stream, keep that property in the metastore and use it here instead
+   * of looking at the SR directly. But this is not compatible with existing streams because
+   * they were created previous to this fix. Those previous streams would fail with INSERT.
+   * The best option was to dynamically look at the SR schema during an INSERT statement.
+   */
+  private static FormatInfo addSerializerMissingFormatFields(
+      final SchemaRegistryClient srClient,
+      final FormatInfo formatInfo,
+      final String topicName,
+      final boolean isKey
+  ) {
+    // Only SR information, such as schema ids and full names are required, which are available
+    // for SR formats.
+    final Format format = FormatFactory.fromName(formatInfo.getFormat());
+    if (!format.supportsFeature(SerdeFeature.SCHEMA_INFERENCE)) {
+      return formatInfo;
+    }
+
+    final Set<String> supportedProperties = format.getSupportedProperties();
+
+    final ImmutableMap.Builder propertiesBuilder = ImmutableMap.builder();
+    propertiesBuilder.putAll(formatInfo.getProperties());
+
+    // The FULL_SCHEMA_NAME allows the serializer to choose the schema definition
+    if (supportedProperties.contains(ConnectProperties.FULL_SCHEMA_NAME)) {
+      if (!formatInfo.getProperties().containsKey(ConnectProperties.FULL_SCHEMA_NAME)) {
+        SchemaRegistryUtil.getParsedSchema(srClient, topicName, isKey).map(ParsedSchema::name)
+            .ifPresent(schemaName ->
+                propertiesBuilder.put(ConnectProperties.FULL_SCHEMA_NAME, schemaName));
+      }
+    }
+
+    // The SCHEMA_ID is used by the serializer to fetch the right SR schema by id
+    if (supportedProperties.contains(ConnectProperties.SCHEMA_ID)) {
+      if (!formatInfo.getProperties().containsKey(ConnectProperties.SCHEMA_ID)) {
+        SchemaRegistryUtil.getSchemaId(srClient, topicName, isKey)
+            .ifPresent(schemaId ->
+                propertiesBuilder.put(ConnectProperties.SCHEMA_ID, String.valueOf(schemaId)));
+      }
+    }
```

Comment:
Why do we need this? This seems to change existing logic of data serialization drastically

Reply:
Because we need the Serde factory to get the Schema and extract the specified schema definition from SR. This happens in `ProtobufSerdeFactory.createSerde()` which calls `SerdeUtils.getAndTranslateSchema()` when the schemaId is present. 

Also, by setting the SCHEMA_ID, the ProtobufSerdeFactory.getConverter() disables the AUTO_REGISTER_SCHEMAS
to prevent failing because multiple schema definitions on SR.

The same happens for Avro and JsonSR factories. I could refactor the interfaces and classes to accept the physical schema extracted, which is what I need. But seems schemaId does the same, so I reuse it.

What other things does the schemaId do that it's for concern?